packages/nvidia_nat_mcp/src/nat/plugins/mcp/client_impl.py

```
@@ -75,103 +87,216 @@ def mcp_client_transport(self, transport: str | None):
+                    logger.info("Cleaning up inactive session client: %s", truncate_session_id(session_id))
+                    client = self._session_clients[session_id]
+                    # Close the client connection
+                    await client.__aexit__(None, None, None)
+                    del self._session_clients[session_id]
+                    del self._session_last_activity[session_id]
+                    # Clean up reference count if it exists
+                    if session_id in self._session_ref_counts:
+                        del self._session_ref_counts[session_id]
+                    logger.info("Cleaned up inactive session client: %s", truncate_session_id(session_id))
+                    logger.info(" Total sessions: %d", len(self._session_clients))
+                except Exception as e:
```

Ruff (0.13.2)

Do not catch blind exception: `Exception`

(BLE001)